### PR TITLE
Add token fetch options property and containerLoaded to events.

### LINKED ISSUE #3894: Better telemetry around TreesLatest token fetch in ODSP driver
In our telemetry, we are leveraging two important telemetry markers. We have a top level "loadContainer" marker, that is measured by the host application. This captures the end to end time to get a container from the runtime. This includes a lot of steps along the way, such as a snapshot fetch and a storage token fetch in the ODSP case. We rely on the ODSP driver's telemetry to get insight into token fetches on the critical path for trees/latest.

One thing we have noticed is that we have way more token fetch calls logged than load container calls. Here are some examples from the past week where we had 6800 container loads, but 9000 TeesLatest_GetToken_end markers. 

Here is the query we use to get the storage token data, including a clause where we filter out the summarizer
Office_Fluid_FluidRuntime_Performance 
| where Data_documentEverHidden == false and Data_eventName == "fluid:telemetry:OdspDriver:TreesLatest_GetToken_end" and Data_clientType == "interactive"
| extend t = Data_duration 

One thing I am not sure about is if there is some retry logic going on here for expired tokens? Is this potentially getting misrepresented in telemetry? I found this code https://github.com/microsoft/FluidFramework/blob/16f58fc65c8b6693a04a75e2b9554656a3a47f5c/packages/drivers/odsp-driver/src/odspUtils.ts#L46

TL;DR:
We as consumers of fluid want a reliable signal for the amount of time spent getting a token. I'm not sure the TreesLatest_GetToken_end marker is giving us this signal, given that we are seeing way more than the expected number of calls.

### PR DESCRIPTION
**Pull request by curtisman**
Fix issue #3894 

Add token fetch options to the perf event to help reasoning over the data.
Also added containerLoaded field to identify events that are part of the load sequence or not.

### CODE REVIEW

**Review by christiango — APPROVED** [review]
Thank you!

File: packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts [hunk 1]
```diff
@@ -142,7 +142,11 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts:142]: Can you please delete StorageTokenRefresh above then? Or do we want to keep it?

**Review by vladsud — APPROVED** [review]

File: packages/loader/container-loader/src/container.ts [hunk 2]
```diff
@@ -480,7 +481,6 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 {
                     eventName: "ContainerClose",
                     sequenceNumber: error.sequenceNumber ?? this._deltaManager.lastSequenceNumber,
-                    loading: !this.loaded,
```
Comment (markfields) [inline packages/loader/container-loader/src/container.ts:483]: Is this always false or something? Why remove it?

Comment (vladsud) [reply packages/loader/container-loader/src/container.ts:483]: Now this same payload is part of every event, so no need to have it here

File: packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts (hunk 1 above)
Comment (curtisman) [reply packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts:142]: Yeah.  If you don't see a use of that either, I will remove it.

Comment (curtisman) [reply packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts:142]: @christiango do you want this separate event to flag perf problems?  Or is it enough to get the information out of the perf event?

IMO, we should remove it.  I generally hate duplicate information.

Comment (christiango) [reply packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts:142]: Let's remove it! I think the other event is sufficient, I am also a fan of less duplication

Comment (curtisman) [reply packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts:140]: done.

**Pull request merged**